components/cam/src/physics/cam/shoc.F90

```diff
@@ -29,6 +29,7 @@ module shoc
 logical :: use_cxx = .true.
 
 real(rtype), parameter, public :: largeneg = -99999999.99_rtype
+real(rtype), parameter, public :: pi = 3.14159265 ! Pi
```

Comment:
Isn't this less than double precision if rtype is double? Does `4*atan(1.0_rtype)` work inside a parameter expression in F90?

Reply:
I must have misunderstood. This is the value used in `physics/share/physics_constants.hpp`. 

`4*atan(1._rtype)` works as a parameter expression.

Reply:
I recommend writing `real(8), parameter :: pi = 3.141592653589793_rtype`. You need rtype to retain double precision and want full precision in the constant.

Reply:
Will this be bfb to `static constexpr Scalar Pi = 3.14159265` in `physics_constants.hpp`? Is this because we use `real(8)` here?

Reply:
I meant to write `rtype` above, not `r8`; I've fixed this. That aside, first, the number of digits to be written should be decided. Then `_rtype` should be suffixed regardless so there is no conversion of the literal to SP before the quantity is assigned to the `real(rtype)`.